src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -151,3 +156,66 @@ def signature_hash_algorithm(self):
             raise UnsupportedAlgorithm(
                 "Signature algorithm OID:{0} not recognized".format(oid)
             )
+
+    @property
+    def extensions(self):
+        # TODO: remove this when we finish landing support for extensions.
+        warnings.warn(
+            "Extension support is not fully implemented. Unsupported "
+            "extensions are IGNORED at this time."
+        )
```

Comment:
I think we should raise an exception when we hit an unsupported critical extension (once we have at least the majority of the RFC 5280 extensions implemented).